specification/redhatopenshift/common-types/resource-management/v3/types.json

```diff
@@ -0,0 +1,695 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "3.0",
+    "title": "Common types"
```

Comment:
>title": "Common types" [](http://example.com/codeflow?start=5&length=22)

Why are you not using the common types file directly?

Reply:
I fixed the issue where we weren't using the common-types files directly, but the migration to the latest common-types files `v6` are resulting in breaking change (cross version) CI errors:
```
The new version has a different format 'uuid' than the previous one ''.
New: [common-types/resource-management/v6/types.json#L591:5](https://github.com/Azure/azure-rest-api-specs/blob/6819c7a7f114669cefeb12959712d03ffd29a287/specification/common-types/resource-management/v6/types.json#L591:5)
parameters.SubscriptionIdParameter
Old: [common-types/resource-management/v3/types.json#L622:5](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v3/types.json#L622:5)
```
There are also similar failures as a result of our most recent API using the v3 types when this PR is now expecting the v6 types:
```
The new version has a different format 'uuid' than the previous one ''.
New: [openshiftclusters/preview/2024-08-12-preview/redhatopenshift.json#L1201:9](https://github.com/Azure/azure-rest-api-specs/blob/6819c7a7f114669cefeb12959712d03ffd29a287/specification/redhatopenshift/resource-manager/Microsoft.RedHatOpenShift/openshiftclusters/preview/2024-08-12-preview/redhatopenshift.json#L1201:9)
paths['/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RedHatOpenShift/openshiftclusters/{resourceName}/secret/{childResourceName}'].patch.parameters
Old: [openshiftclusters/stable/2023-11-22/redhatopenshift.json#L1158:9](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/redhatopenshift/resource-manager/Microsoft.RedHatOpenShift/openshiftclusters/stable/2023-11-22/redhatopenshift.json#L1158:9)
```
Is it safe to disregard the errors from CI or do I need to revert to v3? Is it required to use the latest common-types? I got a previous sign-off with the v3 types, so I assume they are not deprecated.

Reply:
it is not deprecated but strongly suggested to do it.

you can revert to v3 from common-types so that this PR will be merged faster and  keep in mind for updating to latest version for the next PR as there are some minor breaking changes because of it as i can see.

Reply:
Thanks. If we were to stick with v6 for the sake of not needing to do it later, would the breaking changes as a result be OK to merge, or is there some action needed to resolve them?

Reply:
You will have to get an approval from Azure Breaking Change Review Board. - https://aka.ms/brch

Reply:
@razvanbadea-msft I have one more question, and thanks for your help. Let's say we moved forward with v3 for now, and onboard `Identity` definition as-is. What is the path to migrate to v6 in a later API version given that there are new required definitions for MSI in v6 (ManagedServiceIdentity as you said)? Would we be able to keep the `Identity` definition as-is at that point to avoid breaking things on our end or would we need to migrate somehow?

Reply:
You can have a custom one but even in v3 you can use this type in this pr so that when you will upgrade to a new version of common types you can just change the version for it https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v3/managedidentity.json